frontend/src/pages/NewRun.test.tsx

```diff
@@ -50,6 +50,10 @@ describe('NewRun', () => {
   const updateSnackbarSpy = jest.fn();
   const updateToolbarSpy = jest.fn();
 
+  const mockMath = Object.create(global.Math);
+  mockMath.random = () => 0.5;
+  global.Math = mockMath;
```

Comment:
I think the following would be simpler
```
const randomSpy = spyOn(Math, 'random');
randomSpy.mockImplementation(() => 0.5);
```

Reply:
Actually I tried the following and the test run hangs. That's why I used the current code snippet.

const randomSpy = jest.spyOn(Math, 'random');
randomSpy.mockImplementation(() => 0.5);

Not sure what is wrong with the above.

Reply:
I will try to troubleshoot this together, problem with current implementation is that we don't restore the mock.

Reply:
synced offline, seems like the following worked
```
const randomSpy = spyOn(Math, 'random');
randomSpy.mockImplementation(() => 0.5);
```